### PR TITLE
Subdividing Bezier curves in multiple segments

### PR DESCRIPTION
**Pull request by abrock**
I added a function which divides Bezier curves into an arbitrary number of shorter curves at given times (wrt. the original curve).
I also added a toy which allows the user to modify a CubicBezier and choose three times where the bezier should be split. It then plots the resulting curves and their handles.

### CODE REVIEW

**Comment by caryoscelus** [comment]
Looks like you forgot to add toy files.

**Comment by abrock** [comment]
Adding the toy file was prevented by .gitignore, I now added it with "git add -f" but the build still fails and the website wouldn't show me the error message.

**Comment by caryoscelus** [comment]
Yeah, looks like travis is [partly down](https://www.traviscistatus.com/), but failure is probably due to C++11 usage - it was so with [my pr](https://github.com/inkscape/lib2geom/pull/13).

**Comment by abrock** [comment]
Github still shows me "All checks have failed" but the build log shows that it passed: https://travis-ci.org/inkscape/lib2geom/jobs/206673269

**Comment by caryoscelus** [comment]
It clearly shows that it fails, except for mac build, which is probably incorrectly configured to accept c++11 code. (I'm not suggesting that rewriting in legacy c++ will get your patch instantly accepted, though..)

**Comment by abrock** [comment]
Ah, it does three builds, I didn't realize that and thought the last two build had failed and the recent one succeeded. I'll make the code C++03 compatible.

**Comment by abrock** [comment]
Finally, all checks pass. I added "--std=c++03" to CMAKE_CXX_FLAGS in my build, seems that my gcc (6.3.0 20170205 (Debian 6.3.0-6)) has C++11 enabled by default.

File: src/2geom/bezier.cpp [hunk 1]
```diff
@@ -95,6 +95,41 @@ std::pair<Bezier, Bezier> Bezier::subdivide(Coord t) const
     return ret;
 }
 
+std::vector<Bezier> Bezier::subdivide(std::vector<Coord> times_in) const
+{
+    std::vector<Bezier> result;
+    // First we need to sort the times ascending.
+    std::sort(times_in.begin(), times_in.end());
+
+    // Second we filer the times and remove duplicates as well as 0 and 1.
```
Comment (caryoscelus) [inline src/2geom/bezier.cpp]: filer -> filter?

File: src/tests/bezier-subdivision-test.cpp [hunk 2]
```diff
@@ -0,0 +1,363 @@
+/** @file
+ * @brief Unit tests for Affine
+ * Uses the Google Testing Framework
+ *//*
+ * Authors:
+ *   Alexander Brock <Brock.Alexander@web.de>
+ *
+ * Copyright 2010 Authors
```
Comment (caryoscelus) [inline src/tests/bezier-subdivision-test.cpp]: Why 2010?

File: src/2geom/bezier.cpp (hunk 1 above)
Comment (abrock) [reply src/2geom/bezier.cpp]: Yes, thank you.

File: src/tests/bezier-subdivision-test.cpp (hunk 2 above)
Comment (abrock) [reply src/tests/bezier-subdivision-test.cpp]: Because I copied it from somewhere else and didn't pay enough attention, I changed it, thank you.

**Pull request closed without merging**